`opentelemetry-exporter-sender-okhttp:1.52.0` consumed in Maven based applications fails due to missing OkHttp classes

**Describe the bug**
Any application built with Maven that has a (transitive) dependency on `opentelemetry-exporter-sender-okhttp:1.52.0` fails due to missing classes, such as `okhttp3.Callback` and `okhttp3.Interceptor`. I believe this is due to incorrect dependency declarations in OpenTelemetry Java, where dependencies on `com.squareup.okhttp3:okhttp` should instead now be on `com.squareup.okhttp3:okhttp-jvm`. This is due to a change in OkHttp, which since 5.0.0 publishes artifact variants which are only automatically detected in build systems that are Gradle module metadata aware.

**Steps to reproduce**
In a terminal, run this command:
```
docker run --rm -p 16686:16686 -p 4317:4317 jaegertracing/all-in-one:1.69.0
```
In another terminal, prepare the following files:
`./pom.xml`
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>org.example</groupId>
    <artifactId>okhttp3-issue</artifactId>
    <version>0.1</version>

    <dependencies>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
            <version>1.52.0</version>
            <scope>compile</scope>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
            <version>1.52.0</version>
            <scope>compile</scope>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-otlp</artifactId>
            <version>1.52.0</version>
            <scope>runtime</scope>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-assembly-plugin</artifactId>
                <version>3.7.1</version>

                <configuration>
                    <descriptorRefs>
                        <descriptorRef>jar-with-dependencies</descriptorRef>
                    </descriptorRefs>
                </configuration>

                <executions>
                    <execution>
                        <id>make-assembly</id>
                        <phase>package</phase>
                        <goals>
                            <goal>single</goal>
                        </goals>
                    </execution>
                </executions>

            </plugin>
        </plugins>
    </build>

    <properties>
        <maven.compiler.source>21</maven.compiler.source>
        <maven.compiler.target>21</maven.compiler.target>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

</project>
```
`./src/main/java/org/example/OkHttp3Issue.java`
```java
package org.example;

import io.opentelemetry.api.OpenTelemetry;
import io.opentelemetry.api.trace.Span;
import io.opentelemetry.api.trace.Tracer;
import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;

public class OkHttp3Issue {
    public static void main(String[] args) {
        OpenTelemetry openTelemetry = AutoConfiguredOpenTelemetrySdk.builder().build().getOpenTelemetrySdk();
        System.out.println("Initialised");
        String scopeName = OkHttp3Issue.class.getName();
        Tracer tracer = openTelemetry.getTracer(scopeName);
        Span span = tracer.spanBuilder("important work").setAttribute("foo", 42).startSpan();
        span.end();
    }
}
```
Enter the commands:
```
mvn clean package

java -Dotel.traces.exporter=otlp \
    -Dotel.metrics.exporter=none \
    -Dotel.logs.exporter=none \
    -Dotel.resource.attributes=service.name=OkHttp3Issue \
    -cp target/okhttp3-issue-0.1-jar-with-dependencies.jar \
    org.example.OkHttp3Issue
```
**What did you expect to see?**
- The message `Initialised` printed in the terminal
- A single span in the Jaeger UI at http://localhost:16686/search

**What did you see instead?**
- The following error message is printed in the terminal:
```
Exception in thread "main" java.lang.NoClassDefFoundError: okhttp3/Interceptor
        at io.opentelemetry.exporter.sender.okhttp.internal.OkHttpGrpcSenderProvider.createSender(OkHttpGrpcSenderProvider.java:23)
        at io.opentelemetry.exporter.internal.grpc.GrpcExporterBuilder.build(GrpcExporterBuilder.java:236)
        at io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder.build(OtlpGrpcSpanExporterBuilder.java:310)
        at io.opentelemetry.exporter.otlp.internal.OtlpSpanExporterProvider.createExporter(OtlpSpanExporterProvider.java:76)
        at io.opentelemetry.sdk.autoconfigure.internal.SpiHelper.lambda$loadConfigurable$0(SpiHelper.java:76)
        at io.opentelemetry.sdk.autoconfigure.internal.NamedSpiManager.tryLoadImplementationForName(NamedSpiManager.java:51)
        at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
        at io.opentelemetry.sdk.autoconfigure.internal.NamedSpiManager.getByName(NamedSpiManager.java:41)
        at io.opentelemetry.sdk.autoconfigure.SpanExporterConfiguration.configureExporter(SpanExporterConfiguration.java:86)
        at io.opentelemetry.sdk.autoconfigure.SpanExporterConfiguration.configureSpanExporters(SpanExporterConfiguration.java:62)
        at io.opentelemetry.sdk.autoconfigure.TracerProviderConfiguration.configureTracerProvider(TracerProviderConfiguration.java:58)
        at io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.configureSdk(AutoConfiguredOpenTelemetrySdkBuilder.java:538)
        at io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.buildImpl(AutoConfiguredOpenTelemetrySdkBuilder.java:488)
        at io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.build(AutoConfiguredOpenTelemetrySdkBuilder.java:430)
        at org.example.OkHttp3Issue.main(OkHttp3Issue.java:10)
Caused by: java.lang.ClassNotFoundException: okhttp3.Interceptor
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
        ... 15 more
```
- No span is exported to Jaeger


**What version and what artifacts are you using?**
Artifacts: `opentelemetry-exporter-otlp`
Version: 1.52.0
See `pom.xml` above for more details

**Environment**
Compiler:  Temurin-21.0.1+12
OS: Ubuntu 22.04.5 LTS
Maven: 3.9.10

**Additional context**
- The program works as expected using OpenTelemetry 1.51.0 artifacts.
- The version of `com.squareup.okhttp3:okhttp` used by OpenTelemetry 1.52.0 changed from 4.12.0 to 5.1.0. As documented in the [OkHttp release notes](https://square.github.io/okhttp/changelogs/changelog/#version-500) since 5.0.0 OkHttp build artifacts now use Gradle module metadata. Also, OkHttp now publishes separate JVM and Android build artifacts. Applications built using Gradle are not affected, as Gradle automatically uses the appropriate build artifact. However, applications built using Maven are affected as they are not Gradle module metadata aware.
- Overriding the transitive dependency on `com.squareup.okhttp3:okhttp` to instead be `com.squareup.okhttp3:okhttp-jvm` fixes the example repro detailed above. To see this, change the `<dependencies>` section in `./pom.xml` to read as follows:
```xml
    <dependencies>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
            <version>1.52.0</version>
            <scope>compile</scope>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-sdk-extension-autoconfigure</artifactId>
            <version>1.52.0</version>
            <scope>compile</scope>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-otlp</artifactId>
            <version>1.52.0</version>
            <scope>runtime</scope>
            <exclusions>
                <exclusion>
                    <groupId>com.squareup.okhttp3</groupId>
                    <artifactId>okhttp</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.squareup.okhttp3</groupId>
            <artifactId>okhttp-jvm</artifactId>
            <version>5.1.0</version>
        </dependency>
    </dependencies>
```

--- a/dependencyManagement/build.gradle.kts
+++ b/dependencyManagement/build.gradle.kts
@@ -7,25 +7,6 @@ data class DependencySet(val group: String, val version: String, val modules: Li
 val dependencyVersions = hashMapOf<String, String>()
 rootProject.extra["versions"] = dependencyVersions
 
-val DEPENDENCY_BOMS = listOf(
-  // for some reason boms show up as runtime dependencies in license and vulnerability scans
-  // even if they are only used by test dependencies, so not using junit bom here
-  // (which is EPL licensed) or armeria bom (which is Apache licensed but is getting flagged
-  // by FOSSA for containing EPL-licensed)
-
-  "com.fasterxml.jackson:jackson-bom:2.19.2",
-  "com.google.guava:guava-bom:33.4.8-jre",
-  "com.google.protobuf:protobuf-bom:4.31.1",
-  "com.squareup.okhttp3:okhttp-bom:5.1.0",
-  "com.squareup.okio:okio-bom:3.15.0", // applies to transitive dependencies of okhttp
-  "io.grpc:grpc-bom:1.73.0",
-  "io.netty:netty-bom:4.2.3.Final",
-  "io.zipkin.brave:brave-bom:6.3.0",
-  "io.zipkin.reporter2:zipkin-reporter-bom:3.5.1",
-  "org.assertj:assertj-bom:3.27.3",
-  "org.testcontainers:testcontainers-bom:1.21.3",
-  "org.snakeyaml:snakeyaml-engine:2.10"
-)
 
 val autoValueVersion = "1.11.0"
 val errorProneVersion = "2.41.0"
@@ -37,6 +18,27 @@ val opencensusVersion = "0.31.1"
 val prometheusServerVersion = "1.3.10"
 val armeriaVersion = "1.32.5"
 val junitVersion = "5.13.4"
+val okhttpVersion = "5.1.0"
+
+val DEPENDENCY_BOMS = listOf(
+  // for some reason boms show up as runtime dependencies in license and vulnerability scans
+  // even if they are only used by test dependencies, so not using junit bom here
+  // (which is EPL licensed) or armeria bom (which is Apache licensed but is getting flagged
+  // by FOSSA for containing EPL-licensed)
+
+  "com.fasterxml.jackson:jackson-bom:2.19.2",
+  "com.google.guava:guava-bom:33.4.8-jre",
+  "com.google.protobuf:protobuf-bom:4.31.1",
+  "com.squareup.okhttp3:okhttp-bom:$okhttpVersion",
+  "com.squareup.okio:okio-bom:3.15.0", // applies to transitive dependencies of okhttp
+  "io.grpc:grpc-bom:1.73.0",
+  "io.netty:netty-bom:4.2.3.Final",
+  "io.zipkin.brave:brave-bom:6.3.0",
+  "io.zipkin.reporter2:zipkin-reporter-bom:3.5.1",
+  "org.assertj:assertj-bom:3.27.3",
+  "org.testcontainers:testcontainers-bom:1.21.3",
+  "org.snakeyaml:snakeyaml-engine:2.10"
+)
 
 val DEPENDENCIES = listOf(
   "org.junit.jupiter:junit-jupiter-api:${junitVersion}",
@@ -72,6 +74,7 @@ val DEPENDENCIES = listOf(
   "com.google.code.findbugs:jsr305:3.0.2",
   "com.google.guava:guava-beta-checker:1.0",
   "com.sun.net.httpserver:http:20070405",
+  "com.squareup.okhttp3:okhttp-jvm:$okhttpVersion",
   "com.tngtech.archunit:archunit-junit5:1.4.1",
   "com.uber.nullaway:nullaway:0.12.7",
   "edu.berkeley.cs.jqf:jqf-fuzz:1.7", // jqf-fuzz version 1.8+ requires Java 11+

--- a/exporters/otlp/all/build.gradle.kts
+++ b/exporters/otlp/all/build.gradle.kts
@@ -34,7 +34,7 @@ dependencies {
   jmhImplementation("com.linecorp.armeria:armeria")
   jmhImplementation("com.linecorp.armeria:armeria-grpc")
   jmhImplementation("io.opentelemetry.proto:opentelemetry-proto")
-  jmhRuntimeOnly("com.squareup.okhttp3:okhttp")
+  jmhRuntimeOnly("com.squareup.okhttp3:okhttp-jvm")
   jmhRuntimeOnly("io.grpc:grpc-netty")
 }
 
@@ -65,7 +65,7 @@ testing {
             }
           }
 
-          implementation("com.squareup.okhttp3:okhttp")
+          implementation("com.squareup.okhttp3:okhttp-jvm")
           implementation("io.grpc:grpc-stub")
         }
 

--- a/exporters/otlp/all/build.gradle.kts
+++ b/exporters/otlp/all/build.gradle.kts
@@ -64,8 +64,12 @@ testing {
               }
             }
           }
+          if (it.equals("LATEST")) {
+            implementation("com.squareup.okhttp3:okhttp-jvm")
+          } else {
+            implementation("com.squareup.okhttp3:okhttp")
+          }
 
-          implementation("com.squareup.okhttp3:okhttp-jvm")
           implementation("io.grpc:grpc-stub")
         }
 

--- a/exporters/otlp/testing-internal/build.gradle.kts
+++ b/exporters/otlp/testing-internal/build.gradle.kts
@@ -27,7 +27,7 @@ dependencies {
 
   api("io.opentelemetry.proto:opentelemetry-proto")
   api("org.junit.jupiter:junit-jupiter-api")
-  implementation("com.squareup.okhttp3:okhttp")
+  implementation("com.squareup.okhttp3:okhttp-jvm")
   implementation("org.junit.jupiter:junit-jupiter-params")
 
   implementation("com.linecorp.armeria:armeria-grpc-protocol")

--- a/exporters/sender/okhttp/build.gradle.kts
+++ b/exporters/sender/okhttp/build.gradle.kts
@@ -12,7 +12,7 @@ dependencies {
   implementation(project(":exporters:common"))
   implementation(project(":sdk:common"))
 
-  implementation("com.squareup.okhttp3:okhttp")
+  implementation("com.squareup.okhttp3:okhttp-jvm")
 
   compileOnly("io.grpc:grpc-stub")
   compileOnly("com.fasterxml.jackson.core:jackson-core")

--- a/sdk-extensions/jaeger-remote-sampler/build.gradle.kts
+++ b/sdk-extensions/jaeger-remote-sampler/build.gradle.kts
@@ -20,7 +20,7 @@ dependencies {
   implementation(project(":exporters:common"))
   implementation(project(":exporters:sender:okhttp"))
 
-  implementation("com.squareup.okhttp3:okhttp")
+  implementation("com.squareup.okhttp3:okhttp-jvm")
 
   compileOnly("io.grpc:grpc-api")
   compileOnly("io.grpc:grpc-protobuf")

